test/EFCore.PG.FunctionalTests/Query/RangeQueryNpgsqlTest.cs

```diff
@@ -0,0 +1,514 @@
+        /// </summary>
+        public static IEnumerable<object[]> IntegerTheoryData => Enumerable.Range(-10, 10).Select(x => new object[] { x });
+
+        /// <summary>
+        /// Provides theory data for ranges.
+        /// </summary>
+        public static IEnumerable<object[]> RangeTheoryData =>
+            new List<object[]>
+            {
+                // (0,5)
+                new object[] { new NpgsqlRange<int>(0, false, false, 5, false, false) },
+                // [0,5]
+                new object[] { new NpgsqlRange<int>(0, true, false, 5, true, false) },
+                // (,)
+                new object[] { new NpgsqlRange<int>(0, false, true, 0, false, true) },
+                // (,)
+                new object[] { new NpgsqlRange<int>(0, false, true, 5, false, true) },
+                // (0,)
+                new object[] { new NpgsqlRange<int>(0, false, false, 0, false, true) },
+                // (0,)
+                new object[] { new NpgsqlRange<int>(0, false, false, 5, false, true) },
+                // (,5)
+                new object[] { new NpgsqlRange<int>(0, false, true, 5, false, false) }
+            };
+
+        /// <summary>
+        /// Initializes resources for unit tests.
+        /// </summary>
+        /// <param name="fixture">
+        /// The fixture of resources for testing.
+        /// </param>
+        public RangeQueryNpgsqlTest(RangeQueryNpgsqlFixture fixture)
+        {
+            Fixture = fixture;
+            Fixture.TestSqlLoggerFactory.Clear();
+        }
+
+        /// <summary>
+        /// Tests translation for <see cref="NpgsqlRangeExtensions.Contains{T}(NpgsqlRange{T},NpgsqlRange{T})"/>.
+        /// </summary>
+        [Theory]
+        [MemberData(nameof(RangeTheoryData))]
+        public void RangeContainsRange(NpgsqlRange<int> range)
+        {
+            using (RangeContext context = Fixture.CreateContext())
+            {
+                RangeTestEntity[] _ =
+                    context.RangeTestEntities
+                           .Where(x => x.Range.Contains(range))
+                           .ToArray();
+
+                Assert.Contains("WHERE x.\"Range\" @> @__range_0 = TRUE", Fixture.TestSqlLoggerFactory.Sql);
+            }
+        }
+
+        /// <summary>
+        /// Tests inverse translation for <see cref="NpgsqlRangeExtensions.Contains{T}(NpgsqlRange{T}, NpgsqlRange{T})"/>.
+        /// </summary>
+        [Theory]
+        [MemberData(nameof(RangeTheoryData))]
+        public void RangeDoesNotContainRange(NpgsqlRange<int> range)
+        {
+            using (RangeContext context = Fixture.CreateContext())
+            {
+                RangeTestEntity[] _ =
+                    context.RangeTestEntities
+                           .Where(x => !x.Range.Contains(range))
+                           .ToArray();
+
+                Assert.Contains("WHERE NOT (x.\"Range\" @> @__range_0 = TRUE)", Fixture.TestSqlLoggerFactory.Sql);
```

Comment:
Nit: Refactor this out to a single `AssertContainsSql()` which can be called from all the tests.